client/src/components/App.jsx

```diff
@@ -17,8 +18,9 @@ export default class App extends React.Component {
       .then((results) => {
         console.log('AXIOS SUCCESS');
         console.log('AXIOS RESULTS:   ', results);
-        let urlArray = results.data[0].roomImages;
-        this.setState({allImages: urlArray});
+        let allImages = results.data[0].roomImages;
```

Comment:
this 0 is a magic number, just change your API to reflect the fact that you are fetching a single item (findOne?)